cmd/queue/execprobe_test.go

```diff
@@ -63,13 +65,43 @@ func TestProbeQueueNotReady(t *testing.T) {
 		t.Fatalf("Failed to convert port(%s) to int: %v", u.Port(), err)
 	}
 
-	err = probeQueueHealthPath(1, port)
+	err = probeQueueHealthPath(time.Second, port)
+
+	if err == nil || err.Error() != "probe returned not ready" {
+		t.Error("Unexpected not ready error:", err)
+	}
+
+	if queueProbed == 0 {
+		t.Error("Expected the queue proxy server to be probed")
+	}
+}
+
+func TestProbeShuttingDown(t *testing.T) {
+	queueProbed := int32(0)
+	ts := newProbeTestServer(func(w http.ResponseWriter) {
+		atomic.AddInt32(&queueProbed, 1)
+		w.WriteHeader(http.StatusGone)
+	})
+
+	defer ts.Close()
+
+	u, err := url.Parse(ts.URL)
+	if err != nil {
+		t.Fatalf("%s is not a valid URL: %v", ts.URL, err)
+	}
+
+	port, err := strconv.Atoi(u.Port())
+	if err != nil {
+		t.Fatalf("Failed to convert port(%s) to int: %v", u.Port(), err)
+	}
+
+	err = probeQueueHealthPath(time.Second, port)
 
-	if diff := cmp.Diff(err.Error(), "probe returned not ready"); diff != "" {
-		t.Errorf("Unexpected not ready message: %s", diff)
+	if err == nil || err.Error() != "failed to probe: failing probe deliberately for shutdown" {
+		t.Error("Unexpected error:", err)
 	}
 
-	if atomic.LoadInt32(queueProbed) == 0 {
+	if queueProbed == 0 {
```

Comment:
I cant figure out why this is safe 😅 - I know we don't _actually_ write (because the test is green) so there isn't a race, but wouldn't there be if we did?

Reply:
Because by the time we reach the check we stopped probing, right?

Reply:
mmm but there's no guarantee the thread we're on saw it, it could still be in a register on another cpu for example.. there's no memory event here like a channel or anything to create a happens-before

Reply:
In fact, I think in general we can remove all atomic calls out there, given nothing happens in parallel, fwiw.

Reply:
tbh I think they should all've been channels in the first place :) but if they're atomics I think we need a Load

Reply:
(I think the handlers in httptest.NewTestServer run in goroutines?)

Reply:
But you need a request and we don't send requests anymore (since the probe function exited).

Reply:
summary of slack discussion: we need either all the atomics or none - and probably none, as long as the race detector is happy with that :D

Reply:
Removed all atomics.